### PR TITLE
fix: [PL-22558]: Sending base branch in fullSync API

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
##### Summary:

1. Sending base branch in fullSync API
2. Replacing logged-in user name with  SCM user name for 'Using Git credentials of {user}' after product feedback
3. Added tests

Changes are [Tested in PR env](https://pr.harness.io/full-sync-test1/ng/#/account/kmpySmUISimoRrJL6NL73w/cd/orgs/default/projects/sunnyPR_test/setup/git-sync/repos)

##### Jira Links:
https://harness.atlassian.net/browse/PL-22558

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by sunnykesh-harness** [comment]
retrigger titlecheck

**Comment by sunnykesh-harness** [comment]
dockerize

File: src/modules/40-gitsync/components/FullSyncForm/FullSyncForm.tsx [hunk 1]
```diff
@@ -64,6 +64,11 @@ const hasToProcessConfig = (loadingConfig: boolean, repos: GitSyncConfig[]): boo
 
 const getDefaultBranchForPR = (isNew: boolean, defaultBranch?: string): string => (isNew ? defaultBranch || '' : '')
 
+const getBaseBranch = (gitSyncRepos: GitSyncConfig[], repoIdentifier: string) => {
```
Comment (abhinavrastogi-harness) [inline src/modules/40-gitsync/components/FullSyncForm/FullSyncForm.tsx:67]: no return type added on this function?

File: src/modules/40-gitsync/components/FullSyncForm/__tests__/FullSyncForm.test.tsx [hunk 2]
```diff
@@ -78,4 +87,43 @@ describe('Test GitFullSyncForm', () => {
 
     expect(container.querySelector('input[name="prTitle"]')).toBeFalsy()
   })
+
+  test('Should called edit API on save with right payload for old user', async () => {
+    const { getByText } = render(
+      <GitSyncTestWrapper
+        path="/account/:accountId/ci/orgs/:orgIdentifier/projects/:projectIdentifier/admin/git-sync/repos"
```
[the remark this answers is not in the document]
Comment (abhinavrastogi-harness) [reply src/modules/40-gitsync/components/FullSyncForm/__tests__/FullSyncForm.test.tsx:94]: whats the issue? it works as expected. 
see example here: https://github.com/harness/harness-core-ui/blob/master/src/framework/LicenseStore/__test__/LicenseStoreContext.test.tsx#L64

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/40-gitsync/components/FullSyncForm/FullSyncForm.tsx (hunk 1 above)
Comment (sunnykesh-harness) [reply src/modules/40-gitsync/components/FullSyncForm/FullSyncForm.tsx]: corrected.

File: src/modules/40-gitsync/components/FullSyncForm/__tests__/FullSyncForm.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (sunnykesh-harness) [reply src/modules/40-gitsync/components/FullSyncForm/__tests__/FullSyncForm.test.tsx]: Thanks @reetika04 for pointing out the issue while using routes.

**Comment by sunnykesh-harness** [comment]
retrigger jest

File: src/modules/10-common/components/SCMCheck/__tests__/__snapshots__/SCMCheck.test.tsx.snap [hunk 3]
```diff
@@ -64,7 +64,7 @@ exports[`Test SCMCheck SCMCheck should render currentUserLabel using email if na
               style="border-radius: 100%; line-height: 18px; width: 24px; height: 24px; font-size: 10px;"
               tabindex="0"
             >
-              M
+              D
```
Comment (swarajcheguri-harness) [inline src/modules/10-common/components/SCMCheck/__tests__/__snapshots__/SCMCheck.test.tsx.snap:67]: if this is a string, can we use getstrings here, This helps in internationalization

**Review by kapilkumawat86 — APPROVED** [review]

**Review by swarajcheguri-harness — APPROVED** [review]

File: src/modules/40-gitsync/components/FullSyncForm/FullSyncForm.tsx [hunk 4]
```diff
@@ -301,7 +306,7 @@ const FullSyncForm: React.FC<ModalConfigureProps & FullSyncFormProps> = props =>
                   orgIdentifier,
                   projectIdentifier
                 },
-                formData,
+                { ...formData, baseBranch: getBaseBranch(gitSyncRepos, formData.repoIdentifier) },
```
[the remark this answers is not in the document]
Comment (sunnykesh-harness) [reply src/modules/40-gitsync/components/FullSyncForm/FullSyncForm.tsx:309]: Repo will always have a base branch.

**Pull request merged**